### ISSUE TITLE
Implement PR target attribute

### ISSUE THREAD

**Issue by violette** [renovatebot/renovate#10938]
### Discussed in https://github.com/renovatebot/renovate/discussions/10936

As explained in the discussion mentionned above, as a user, I'd like to run the renovate bot using the self-hosted configuration from my fork, create branch on my fork and open pull requests in a my original repository.

As a developer in my company, we are using stash and each dev has its own fork. Modifiying the original master is not authorized (no branch creation, no direct modification => the bot cannot be installed, it must be launched by a user).
The bot PRs must be created in the fork and the destination should be configurable: the original master.
It seems the same process is applied in GitHub(forkMode).

The value of the pull requests target should be a configuration attribute, which is defaulting to the current behavior.

**Comment by rarkins** [comment]
It's quite more complex and needs someone with familiarity with Renovate to fully specify. e.g.

- Will you commit a config to your fork, or can it be added to the parent master?
- Will you have onboarding PRs?
- Are you running on more than one repo at a time?

**Comment by violette** [comment]
Hi,

> Will you commit a config to your fork, or can it be added to the parent master?

The configuration can be commited to a fork(remote), no modification on the parent(origin master). 

> Will you have onboarding PRs?

No.

>Are you running on more than one repo at a time?

Targets only one repo at the time. The bot(self-hosted) is launched from the fork. Branches are created on the fork but the destination (merge) of the pull-request is the original master repository.

**Comment by rarkins** [comment]
How do you prevent the renovate config on the fork from being included in the pull request?

**Comment by violette** [comment]
I wrote 'can', but preferably the configuration is a local file. 
This said, I looked at https://support.atlassian.com/bitbucket-cloud/docs/exclude-files-from-pull-request-diffs/ and it seems that stash support ignore files, similar to the .gitignore file

**Comment by rarkins** [comment]
I don't think it means that the files are excluded from the PR, only that they're excluded from the view.

**Comment by rarkins** [comment]
It's possible to have no-config renovate though.

Next question: does stash return the parent repo to us during init? @viceice wasn't that confirmed just recently?

**Comment by viceice** [comment]
> Next question: does stash return the parent repo to us during init? @viceice wasn't that confirmed just recently?

It was `bitbucket-server` not `bitbucket-cloud`. @violette Are you using `bitbucket-server` (former name was stash) or the hosted gitbucket cloud at https://bitbucket.org?

**Comment by violette** [comment]
@viceice It's bitbucket-server (stash)

**Comment by viceice** [comment]
> @viceice It's bitbucket-server (stash)

So your docs link is wrong, it's for bitbucket cloud 😉 

@rarkins Yes, we can know the upstream / parent project if this is a fork. So we should be able to create a pr upstream (via config flag). but that needs comunity contribution as you need a server for testing 🙃

**Comment by violette** [comment]
Indeed. I removed the comment. :+1: 
I'd be happy to test the patch if you explain me how to build the docker image locally from the build.

**Comment by viceice** [comment]
@violette You don't need the docker image, you can simply run it from local shell (event windows or macos should work

We won't create the patch / feature, it must come from comunity.
